src/Util/Lock/SemaphoreLockDriver.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Friendica\Util\Lock;
+
+class SemaphoreLockDriver implements ILockDriver
+{
+	/**
+	 * @var array stored semaphores
+	 */
+	private static $semaphore = [];
+
+	/**
+	 * @brief Creates a semaphore key
+	 *
+	 * @param string $key Name of the lock
+	 *
+	 * @return integer the semaphore key
+	 */
+	private static function semaphoreKey($key)
+	{
+		$temp = get_temppath();
+
+		$file = $temp.'/'.$key.'.sem';
+
+		if (!file_exists($file)) {
+			file_put_contents($file, $key);
+		}
+
+		return ftok($file, 'f');
+	}
+
+	/**
+	 *
+	 * @brief Sets a lock for a given name
+	 *
+	 * @param string $key The Name of the lock
+	 * @param integer $timeout Seconds until we give up
+	 *
+	 * @return boolean Was the lock successful?
+	 */
+	public function acquireLock($key, $timeout = 120)
+	{
+		// The second parameter for "sem_acquire" doesn't exist before 5.6.1
+		if (function_exists('sem_get') && version_compare(PHP_VERSION, '5.6.1', '>=')) {
```

Comment:
I will bump the version requirement for Friendica to PHP 5.6.1 with my config file update in the next version, so you probably can remove this version checking code.

Reply:
Why bump it from 5.6 to 5.6.1?

Reply:
To get typed values when parsing the INI configuration strings.

Reply:
How will it break if it expects an integer but gets a string then?

Reply:
I don't know and I really don't want to find out. The `.htconfig.php` file used typed values (integers mostly) and I'd like to keep it this way with `local.ini.php`.

Reply:
But you have to convert the values to the correct type anyhow right? Or is this done by a library?

Reply:
`parse_ini_string()` does it by itself when provided with an additional parameter that was only introduced in PHP 5.6.1